types/node/index.d.ts

```diff
@@ -756,20 +756,23 @@ declare module "http" {
         write(str: string, encoding?: string, cb?: Function): boolean;
         write(str: string, encoding?: string, fd?: string): boolean;
 
-        writeContinue(): void;
-        writeHead(statusCode: number, reasonPhrase?: string, headers?: any): void;
-        writeHead(statusCode: number, headers?: any): void;
-        statusCode: number;
-        statusMessage: string;
+        addTrailers(headers: any): void;
+        finished: boolean;
+        getHeader(name: string): string;
+        getHeaderNames(): string[]
+        getHeaders(): any
```

Comment:
Your example is correct, but it doesn't disprove that the interface doesn't inherit from Object.

You can easily test this. Create an object that prototypically inherits from null. Assign an interface to this object and you will see that TypeScript thinks that the object has a hasOwnProperty method, but in reality it doesn't. The reason for this discrepancy is that interface extends/inherits from Object by default.